particles/Tutorial/Javascript/Demo/source/Game.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+'use strict';
+const resetMove = {type: 'reset', move: -1};
+const clearMove = {type: 'clear', move: -1};
+
+/* global defineParticle */
+defineParticle(({SimpleParticle, html, log}) => {
+
+  return class extends SimpleParticle {
+
+    get template() {
+      return html`
+      <span>It is your turn <span>{{name}}</span>, playing as <span>{{avatar}}</span>.
+      <div slotid="gameSlot"></div>
+      <div hidden={{hideCongrats}}>Congratulations <span>{{winnerName}}</span>!</div>
+    `;
+    }
+
+    shouldRender({gameState, playerOne, playerTwo}) {
+      return gameState && playerOne && playerTwo;
+    }
+
+    render({gameState, playerOneMove, playerTwoMove, playerOne, playerTwo, move, event}) {
+      const toReturn = {hideCongrats: true};
+      if (!move) {
+        this.set('move', resetMove);
+        return toReturn;
```

Comment:
Returning after each conditional is not DRY, and also prevents this function from doing more than one task per iteration, which can impair performance.

`render` is intended only for doing calculations that end up in the return value. Use `update` for handle or state logic that doesn't directly affect the _render-model_ (the thing returned from `render`).